app/lib/katello/resources/cdn/katello_cdn.rb

```diff
@@ -72,16 +76,18 @@ def repository_url(content_label:)
           query_params = params.map { |key, value| "#{key}=#{value}" }
           url = "/katello/api/v2/organizations/#{organization['id']}/repositories?#{query_params.join('&')}"
           response = get(url)
-          repository = JSON.parse(response)['results'].first
-
+          repository = JSON.parse(response)['results'].find {|r| r['arch'] == arch && r['major'] == major && r['minor'] == minor}
```

Comment:
add a unit test for this. I dont see a test file but just a simple test instantiating this object and making sure the results the parsed correctly will be great.